src/library/basetracktablemodel.cpp

```diff
@@ -156,9 +120,23 @@ BaseTrackTableModel::BaseTrackTableModel(
 
 void BaseTrackTableModel::initTableColumnsAndHeaderProperties(
         const QStringList& tableColumns) {
+    // This function might be invoked repeatedly and not only once during construction.
```

Comment:
The method is called repeatedly when activating a crate view. The debug assertions that should verify that this method is called at most once triggered:

```
#5  0x00000000006e4556 in mixxx_debug_assert (assertion=<optimized out>, file=<optimized out>, line=<optimized out>, function=<optimized out>)
    at /home/uk/volumes/Build/cpp/mixxx/src/util/assert.h:9
#6  0x0000000000bbb08e in BaseTrackTableModel::initTableColumnsAndHeaderProperties (this=0x302a5eb8, tableColumns=...)
    at /home/uk/volumes/Build/cpp/mixxx/src/library/basetracktablemodel.cpp:142
#7  0x0000000000b883c0 in BaseSqlTableModel::setTable (this=this@entry=0x302a5eb8, tableName=..., idColumn=..., tableColumns=..., trackSource=...)
    at /home/uk/volumes/Build/cpp/mixxx/src/library/basesqltablemodel.cpp:400
#8  0x0000000000e26586 in CrateTableModel::selectCrate (this=this@entry=0x302a5eb8, crateId=..., crateId@entry=...) at /usr/include/qt6/QtCore/qarraydata.h:59
#9  0x0000000000e05e87 in CrateFeature::activateChild (this=0x302a5e40, index=...) at /home/uk/volumes/Build/cpp/mixxx/src/library/trackset/crate/cratefeature.cpp:310
#10 0x00007ffff07fc8f2 in void doActivate<false>(QObject*, int, void**) () at /lib64/libQt6Core.so.6
#11 0x00007ffff5a7f49c in QAbstractItemView::clicked(QModelIndex const&) () at /lib64/libQt6Widgets.so.6
```

See also: https://github.com/mixxxdj/mixxx/commit/bb0495e7f1cfb2fc3fd327b90f8bc8a52ff79159

Reply:
Doesn't cause issues. Just unexpected and inefficient.

Reply:
https://github.com/mixxxdj/mixxx/pull/13682/commits/bf4af50f236d856eb5af45d5dfc8e49508a7901c

Reply:
I have restored the original optimization.

Just want to mention that this is not correct, because `initHeaderProperties()` is a virtual function. An override in a derived class might depend on the actual contents of `ColumnCache` even if the implementation in the base class does not.

Reply:
This commit should restore the original optimization: https://github.com/mixxxdj/mixxx/pull/13682/commits/dc287ebcf6151355e42ac705b456706ad4343ae3